source/common/upstream/upstream_impl.cc

```diff
@@ -660,6 +661,7 @@ ClusterInfoImpl::ClusterInfoImpl(
       drain_connections_on_host_removal_(config.drain_connections_on_host_removal()),
       warm_hosts_(!config.health_checks().empty() &&
                   common_lb_config_.ignore_new_hosts_until_first_hc()),
+      http_protocol_options_(config.common_http_protocol_options()),
```

Comment:
This is always going to initialize the optional, and fill it with default values. I don't think this is what you intend. Please add an explicit test to cover this case.

Reply:
@mattklein123 It may be good to rewrite with 
```cpp
http_protocol_options_(config.has_common_http_protocol_options() ? absl::make_optional<envoy::api::v2::core::HttpProtocolOptions>(config.common_http_protocol_options()) : absl::nullopt)
```
this is what I was intended. I was overlooked on this initialize section. so I think that It may not needed to add test case.